lib/pbench/test/unit/server/query_apis/test_controllers_list.py

```diff
@@ -99,11 +98,11 @@ def test_query(
         # don't expect success for an "invalid" authentication, for a different
         # user, or for an invalid username.
         if (
-            not user
-            or user == "no_user"
-            or build_auth_header["header_param"] == "valid"
+            user == "no_user" or build_auth_header["header_param"] == "valid"
         ) and user != "badwolf":
             expected_status = HTTPStatus.OK
+        elif user == "drb" and build_auth_header["header_param"] == "valid_admin":
+            expected_status = HTTPStatus.OK
```

Comment:
These patterns show up repeatedly, and I keep wondering about trying to refactor them out into common code. I'm not quite sure that's feasible, but we're certainly creating a lot of extra work for ourselves every time we have to tweak them.

Part of this is just dealing with the bandwidth between our inflexible fixtures and the tests, and pulling some of the setup out of fixtures and into support methods with real parameters and returns would probably help.

Despite this rambling, I don't really expect you to try to solve that in this PR ... we really do, however, need to think about a massive unit test rearchitecture to clean things up. (Including dumping the live DB and Flask calls and moving to a real unit test system.)

Reply:
> Including dumping the live DB and Flask calls and moving to a real unit test system.

While I'm in favor of explicitly mocking the DB calls, I think getting rid of Flask will just trade one kind of mocking for another.  (That is, if I'm not too confused, Flask provides nice facilities for mocking its `requests` package, and I don't _think_ we want to give that up.)